fix: allOf false positives in lint (#3006)
Redocly/redocly-cli: .changeset/rfc7807-allof-false-positives.md

---
'@redocly/openapi-core': patch
'@redocly/cli': patch
---

> Fixed the `operation-4xx-problem-details-rfc7807` rule to resolve `$ref`s and composed schemas, so the `type` and `title` properties inherited through `allOf` are no longer reported as missing.

Reviewer: ```suggestion
Fixed an issue where the `operation-4xx-problem-details-rfc7807` rule incorrectly reported the `type` and `title` properties inherited through `allOf` as missing.
```

---
'@redocly/openapi-core': patch
'@redocly/cli': patch
---

Fixed the `operation-4xx-problem-details-rfc7807` rule to resolve `$ref`s and composed schemas, so the `type` and `title` properties inherited through `allOf` are no longer reported as missing.

> **Note**: the rule now also reports problem schemas that define no properties at all, which previously escaped the check.

Reviewer 2: This seems wrong. The rule was designed to catch missing properties in the first place.
